### PR TITLE
python 3 compatibility

### PR DESCRIPTION
**Pull request by gutfeeling**
Hi Myle, 

twokenize.py should be Python 2 and 3 compatible with these commits. Sorry about the last pull request, I found a bug after creating that pull request. I have fixed it now. 

One more thing. The name of the repo has hyphens in it. When I clone it and try to make it into a package by putting a **init**.py file in the repo, Python complains since it doesn't like hyphens in package names. Maybe it's a good idea to replace the hyphens with underscores?

-Dibya

### CODE REVIEW

File: twokenize.py [hunk 1]
```diff
@@ -19,12 +19,21 @@
...
 
 import operator
 import re
-import HTMLParser
+import sys
+
+try:
+    from html.parser import HTMLParser
+except ImportError:
+    from HTMLParser import HTMLParser
```
Comment (myleott) [inline twokenize.py:31]: It'd be clearer if we could use sys.version.info here instead of try/except.

File: twokenize.py [hunk 2]
```diff
@@ -285,15 +293,22 @@ def tokenize(text):
...
-    text = HTMLParser.HTMLParser().unescape(text)
+    #http://stackoverflow.com/questions/2360598
+    #/how-do-i-unescape-html-entities-in-a-string-in-python-3-1
+    if sys.version.info[0] > 2:
+        if sys.version.info[0] == 3 and sys.version.info[1] <= 3:
+            text = HTMLParser().unescape(text)
+        else:
+            text = html.unescape(text)
+    else:
+        text = HTMLParser().unescape(text)
```
Comment (myleott) [inline twokenize.py:304]: can this be simplified to:

```
if (sys.version.info[0] < 3 or
        (sys.version.info[0] == 3 and sys.version.info[1] <= 3)):
    text = HTMLParser().unescape(text)
else:
    text = html.unescape(text)
```

File: twokenize.py [hunk 3]
```diff
@@ -285,15 +293,22 @@ def tokenize(text):
...
     return text
 
 # This is intended for raw tweet text -- we do some HTML entity unescaping before running the tagger.
-# 
+#
 # This function normalizes the input text BEFORE calling the tokenizer.
 # So the tokens you get back may not exactly correspond to
 # substrings of the original text.
 def tokenizeRawTweetText(text):
```
Comment (myleott) [inline twokenize.py:312]: Both before and after your changes, this code works poorly on non-unicode text. What do you think about adding `assert isinstance(text, unicode)` to the top of this function and also to the top of the `tokenize` function?

**Review by myleott — CHANGES_REQUESTED** [review]
Sorry for taking so long to get to this, but I wanted to test it a bit first and unfortunately I don't have access to my original test set.

Anyway, looks pretty good, but I have a few suggestions/ideas below.

**Comment by gutfeeling** [comment]
Hi Myle,

Thanks for your response.

1. We could use sys.version.info to import the correct module, but I used
try/except because PSF recommends the use of try/import whenever possible
<https://docs.python.org/3/howto/pyporting.html#use-feature-detection-instead-of-version-detection>
.

2. Agree with your second suggestion.

3. If you don't want to support non unicode input, I think raising a
TypeError or ValueError is better than raising an AssertionError.

Cheers,
Dibya

**Comment by myleott** [comment]
1. Interesting, didn't know about that recommendation, makes sense.

3. Sure, let's raise Type or ValueError (your choice).

**Pull request merged**